pkg/cloud/services/gc/service.go

```diff
@@ -88,3 +106,23 @@ func (fn *ResourceCleanupFuncs) Execute(ctx context.Context, resources []*AWSRes
 
 	return nil
 }
+
+// ResourceCollectFunc is a function type to collect resources for a specific AWS service type.
+type ResourceCollectFunc func(ctx context.Context) ([]*AWSResource, error)
+
+// ResourceCollectFuncs is a collection of ResourceCollectFunc.
+type ResourceCollectFuncs []ResourceCollectFunc
+
+// Execute will execute all the defined collect functions against the aws resources.
+func (fn *ResourceCollectFuncs) Execute(ctx context.Context) ([]*AWSResource, error) {
```

Comment:
Since we aren't modifying the custom slice type, you don't need a pointer receiver here. Incidentally, this will also fix having to dereference the type in the below range loop. Same goes for the other type. :)

Reply:
Thanks @Skarlso for the valuable comment.
I'll update it as well as previous existing `fn *ResourceCleanupFuncs` https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/gc/service.go#L82 . Is this oK?

Reply:
Yep 👍